### PR TITLE
feat: Support reexporting more kinds of symbols

### LINKED ISSUE #58: Symbol cannot be exported twice
### Example:
```typescript
export function foo() {}
export { foo as bar };
```
`deno doc` should show two exports, `function foo()` and `function bar()`. However, this currently only shows `foo()`.

I think this is part of the issue in #57.

### LINKED ISSUE #61: Doesn't detect reexported imports
### Example
```typescript
// foo.ts
export const foo = "foo";
```
```typescript
// mod.ts
import { foo } from "./foo.ts";

export { foo };
```
`deno doc` should show `foo` as an export of `mod.ts`, but doesn't.

### LINKED ISSUE #62: Doesn't detect reexported default exports
### Example
```typescript
function foo() {}
export default foo;
```
`deno doc` should show `foo` as the default export, but shows nothing.

### PR DESCRIPTION
**Pull request by Liamolucko**
This is aiming to fix #58, #61 and #62. I've found myself having to re-architect some stuff for this, and I don't really want to make multiple PRs which depend on each other. So, I'm just combining them into one.

For GitHub's sake:
Fixes #61. Fixes #62.

### CODE REVIEW

**Comment by Liamolucko** [comment]
For GitHub's sake:
Fixes #61. Fixes #62

**Comment by bartlomieju** [comment]
@Liamolucko please rebase

File: src/tests.rs [hunk 1]
```diff
@@ -231,20 +231,6 @@ export function fooFn(a: number) {
...
-      "location": {
-        "filename": "file:///test.ts",
-        "line": 3,
-        "col": 0
-      },
-      "jsDoc": null,
-      "importDef": {
-        "src": "file:///reexport.ts",
-        "imported": "fizz",
-      }
```
Comment (bartlomieju) [inline src/tests.rs:247]: Why are those removed from the test?

**Review by bartlomieju — COMMENTED** [review]
@Liamolucko sorry for slow review. Nice work! In general looks good to me I have a few questions about removal of some stuff. Also please draft a commit message

File: src/node.rs [hunk 2]
```diff
@@ -46,8 +46,6 @@ pub enum ReexportKind {
   All,
   /// export * as someNamespace from "./path/to/module.js";
   Namespace(String),
-  /// export default from "./path/to/module.js";
-  Default,
```
[the remark this answers is not in the document]
Comment (Liamolucko) [reply src/node.rs:50]: I originally removed it because it's invalid syntax, but I didn't realise it was a proposal - I'll put it back.

[the remark this answers is not in the document]
Comment (Liamolucko) [reply src/node.rs:50]: Actually, it's probably simpler to reinterpret it as a named export - `export foo from "./foo.ts"` is the same as `export { default as foo } from "./foo.ts"`.

File: src/tests.rs (hunk 1 above)
Comment (Liamolucko) [reply src/tests.rs:247]: I had originally intended to entirely replace the existing import handling with handling them as reexports, but I just realised they're still needed for linking to types which aren't reexported by the module but are still included in the interface. I'll put it back.

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks @Liamolucko
